data/transactions/logic/eval.go

```diff
@@ -355,58 +356,75 @@ func copyWithClearAD(txgroup []transactions.SignedTxnWithAD) []transactions.Sign
 	return copy
 }
 
-// NewEvalParams creates an EvalParams to use while evaluating a top-level txgroup
-func NewEvalParams(txgroup []transactions.SignedTxnWithAD, proto *config.ConsensusParams, specials *transactions.SpecialAddresses) *EvalParams {
-	apps := 0
+// NewSigEvalParams creates an EvalParams to be used while evaluating a group's logicsigs
+func NewSigEvalParams(txgroup []transactions.SignedTxn, proto *config.ConsensusParams, ls LedgerForSignature) *EvalParams {
+	lsigs := 0
 	for _, tx := range txgroup {
-		if tx.Txn.Type == protocol.ApplicationCallTx {
-			apps++
+		if !tx.Lsig.Blank() {
+			lsigs++
 		}
 	}
 
-	// Make a simpler EvalParams that is good enough to evaluate LogicSigs.
-	if apps == 0 {
-		return &EvalParams{
-			TxnGroup: txgroup,
-			Proto:    proto,
-			Specials: specials,
+	var pooledLogicBudget *int
+	if lsigs > 0 { // don't allocate if no lsigs
+		if proto.EnableLogicSigCostPooling {
+			pooledLogicBudget = new(int)
+			*pooledLogicBudget = len(txgroup) * int(proto.LogicSigMaxCost)
 		}
 	}
 
-	minAvmVersion := ComputeMinAvmVersion(txgroup)
+	withADs := transactions.WrapSignedTxnsWithAD(txgroup)
+	return &EvalParams{
+		runMode:              ModeSig,
+		TxnGroup:             withADs,
+		Proto:                proto,
+		minAvmVersion:        computeMinAvmVersion(withADs),
+		SigLedger:            ls,
+		PooledLogicSigBudget: pooledLogicBudget,
+	}
+}
+
+// NewAppEvalParams creates an EvalParams to use while evaluating a top-level txgroup.
+func NewAppEvalParams(txgroup []transactions.SignedTxnWithAD, proto *config.ConsensusParams, specials *transactions.SpecialAddresses) *EvalParams {
```

Comment:
now there's the same parameter name in two next-door functions that have different types.

Reply:
It seemed weird to demand that lsig ep construction callers wrap the SignedTxns into SignedTxnWIthADs before calling, as ADs are about the stateful part of evaluation.  Those callers don't know about ADs.

On the other hand, for app evaluation, the callers have SignedTxnsWithAD, and I didn't want to make them copy them into a plain `[]SignedTxn`, since I would then have to re-copy anyway, since the evaluator needs the AD space to work with.